src/blocks/folder-blocks/infinite-canvas/index.tsx

```diff
@@ -0,0 +1,282 @@
+      if (!wrapperElement?.current) return;
+      wrapperElement?.current.removeEventListener("wheel", onWheel);
+    };
+  }, []);
+
+  const { isDragging } = useDrag(wrapperElement, pan, (newPan) => {
+    const boundedDimensions = [
+      Math.min(Math.max(newPan[0], -width), 0),
+      Math.min(Math.max(newPan[1], -height), 0),
+    ] as Dimensions;
+    setPan(boundedDimensions);
+  });
+
+  const [items, setItems] = useState<ItemType[]>([]);
+  const [isDirty, setIsDirty] = useState(false);
+
+  const [fileContents, setFileContents] = useState<Record<string, string>>({});
+  const files = useMemo(
+    () => getNestedFileTree(tree)[0].children.filter((d) => d.type === "blob"),
+    [tree]
+  );
+
+  const getFileContents = async (path: string) => {
+    const url = `/repos/${context.owner}/${context.repo}/contents/${path}`;
+    const data = await onRequestGitHubData(url);
+    if (!data.content) {
+      return;
+    }
+    const decodedContent = Buffer.from(data.content, "base64").toString("utf8");
+    return decodedContent;
+  };
+
+  useEffect(() => {
+    items.forEach(async (item) => {
+      const path = item.path;
+      if (!path) return;
+      if (fileContents[path]) return;
+      setFileContents((fileContents) => ({
+        ...fileContents,
+        [path]: "Loading...",
+      }));
+      const content = await getFileContents(path);
+      if (!content) return;
+      setFileContents((fileContents) => ({ ...fileContents, [path]: content }));
+    });
+  }, [items]);
+
+  useEffect(() => {
+    setItems(metadata.items || placeholderItems);
+  }, [metadata]);
+
+  const [blockOptions, setBlockOptions] = useState<any[]>([]);
+
+  const getBlocks = async () => {
+    const url = "https://blocks-marketplace.githubnext.com/api/blocks";
+    const res = await fetch(url).then((res) => res.json());
+    const exampleBlocks =
+      res.find((d: any) => d.full_name === "githubnext/blocks-examples")
+        ?.blocks || [];
+    setBlockOptions(
+      exampleBlocks
+        .map((block: any) => ({
+          ...block,
+          owner: "githubnext",
+          repo: "blocks-examples",
+        }))
+        .map((block: Block) => ({
+          ...block,
+          key: getBlockKey(block),
+        }))
+    );
+  };
+  useEffect(() => {
+    getBlocks();
+  }, []);
+
+  return (
+    <div className="wrapper">
+      {/* add new item buttons */}
+      <div className="position-absolute top-2 left-2 z-10 flex flex-col space-y-2">
+        {/* add text */}
+        <button
+          className="btn"
+          onClick={() => {
+            const newItems = [
+              ...items,
+              {
+                type: "text",
+                text: "Hello World",
+                position: [
+                  width / 2 - defaultDimensions[0] / 2,
+                  height / 2 - defaultDimensions[1] / 2,
+                ],
+                dimensions: defaultDimensions,
+              } as ItemType,
+            ];
```

Comment:
`const newItems: ItemType[] = ...` should work, the parts of the context type are propagated down the expression.